community/kernel/src/main/java/org/neo4j/graphdb/GraphDatabaseService.java

```diff
@@ -183,10 +187,55 @@
      * space).
      *
      * @return all relationship types in the underlying store
-     * @deprecated this operation can be found in {@link GlobalGraphOperations} instead.
      */
-    @Deprecated
-    Iterable<RelationshipType> getRelationshipTypes();
+    ResourceIterable<RelationshipType> getRelationshipTypes();
+
+    /**
+     * Returns all relationship types currently in use in the underlying store. Relationship types are
+     * added to the underlying store the first time they are used in a successfully committed
+     * {@link Node#createRelationshipTo node.createRelationshipTo(...)}. Note that this method is
+     * guaranteed to return all known relationship types, and it guarantees that it won't return
+     * "historic" relationship types that no longer have any relationships in the graph.
+     *
+     * @return all relationship types in use in the underlying store
+     */
+    ResourceIterable<RelationshipType> getAllRelationshipTypesInUse();
```

Comment:
I find it weird to have two methods `getRelationshipTypes` and `getAllRelationshipTypesInUse`. What am I supposed to do with all unused relationship types? I think we should have one method `getAllRelationshipTypes` for consistency with the other `getAll...` methods. Or simply just keep the `getRelationshipTypes` method and remove `getAllRelationshipTypesInUse`